tasks/transform_conformance/tests/babel-plugin-transform-react-jsx/test/fixtures/refresh/generates-valid-signature-for-exotic-ways-to-call-hooks/output.js

```diff
@@ -1,6 +1,6 @@
-var _s2 = $RefreshSig$();
 import FancyHook from "fancy";
 import { jsxs as _jsxs } from "react/jsx-runtime";
+var _s2 = $RefreshSig$();
```

Comment:
The plugin is not part of Babel Official and all tests we have copied over on our codebase, so we can control anything. Do you think we can do this?

Reply:
Ah OK. Reason I was asking is that I'm thinking that we could put our transformer through its paces by running it on tons of packages from Github, and check we produce same output as Babel. That would be a very strong testing regime, and give extremely high confidence we have no bugs.

But I guess if Babel doesn't implement React Refresh, we'd need to turn off that transform. So yes, the statement order doesn't matter here.

Out of interest, where did we copy the tests from? That should probably be noted in a comment.

Reply:
https://github.com/facebook/react/blob/main/packages/react-refresh/src/__tests__/ReactFreshBabelPlugin-test.js

Oh, I thought I wrote. I will add the source somewhere later

> But I guess if Babel doesn't implement React Refresh, we'd need to turn off that transform. So yes, the statement order doesn't matter here.

Yes, we have disabled this plugin in all tests except those tests we copied from `react` because this plugin even never passed the Idempotence Test.

Reply:
OK cool. I don't think Refresh transform *should* to pass idempotence test. It's not a downlevelling transform.